airflow/providers/common/sql/hooks/sql.py

```diff
@@ -515,3 +516,52 @@ def test_connection(self):
             message = str(e)
 
         return status, message
+
+    def get_openlineage_database_info(self, connection):
+        """
+        Should return database specific information needed
+        to generate and parse lineage metadata.
+        This includes information helpful for constructing information schema query
+        and creating correct namespace.
+
+        :param connection: Airflow connection to reduce calls of `get_connection` method
+        :return: A :class:`airflow.providers.openlineage.sqlparser.DatabaseInfo` instance.
```

Comment:
```suggestion
```

Reply:
Why so?

Reply:
This can be expressed in the type annotation instead (so you need to add a `-> DatabaseInfo` if this is deleted)

Reply:
As it's empty method that may be optionally implemented, is it enough to annotate it with `-> DatabaseInfo | None`? I'm no expert in static typing.